apps/src/netsim/levels.js

```diff
@@ -1,12 +1,279 @@
+ *           that should be active by default, which depends on which tabs
+ *           you have enabled.
+ *
+ * @property {EncodingType[]} showEncodingControls
+ *
+ * @property {EncodingType[]} defaultEnabledEncodings
+ *
+ * @property {boolean} showDnsModeControl - Whether the DNS mode controls will
+ *           be available to the student.
+ *
+ * @property {DnsMode} defaultDnsMode - Which DNS mode the simulator should
+ *           initialize into.
+ */
 
 /*
  * Configuration for all levels.
  */
 var levels = module.exports = {};
 
-levels.netsim_demo = {
-  'freePlay': true
+/**
+ * A default level configuration so that we can define the others by delta.
+ * This default configuration enables everything possible, so other configs
+ * should start with this one and disable features.
+ * @type {NetSimLevelConfiguration}
+ */
+var defaultLevel = {
+
+  // Lobby configuration
+  showClientsInLobby: true,
+  showRoutersInLobby: true,
+  showAddRouterButton: true,
+
+  // Tab-panel control
+  showTabs: [
+    NetSimTabType.INSTRUCTIONS,
+    NetSimTabType.MY_DEVICE,
+    NetSimTabType.ROUTER,
+    NetSimTabType.DNS
+  ],
+  defaultTabIndex: 0,
+
+  // Instructions tab and its controls
+  // Nothing here yet!
+
+  // "My Device" tab and its controls
+  showEncodingControls: [
+    EncodingType.BINARY,
+    EncodingType.A_AND_B,
+    EncodingType.HEXADECIMAL,
+    EncodingType.DECIMAL,
+    EncodingType.ASCII
+  ],
+  defaultEnabledEncodings: [
+    EncodingType.ASCII,
+    EncodingType.BINARY
+  ],
+
+  // Router tab and its controls
+  // Nothing here yet!
+
+  // DNS tab and its controls
+  showDnsModeControl: true,
+  defaultDnsMode: DnsMode.NONE
 };
+
+/**
+ * Just uses the default level configuration.
+ * @type {NetSimLevelConfiguration}
+ */
+levels.netsim_demo = defaultLevel;
+
+/**
+ * Variant 1 base level
+ * @type {NetSimLevelConfiguration}
+ */
+levels.variant1 = utils.extend(defaultLevel, {
+  showAddRouterButton: false,
+
+  showTabs: [NetSimTabType.INSTRUCTIONS],
+
+  defaultEnabledEncodings: [EncodingType.A_AND_B]
+});
+
+/**
+ * Variant 2 base level
+ * @type {NetSimLevelConfiguration}
+ */
+levels.variant2 = utils.extend(defaultLevel, {
+  showAddRouterButton: false,
+  showTabs: [NetSimTabType.INSTRUCTIONS, NetSimTabType.MY_DEVICE],
+  showEncodingControls: [EncodingType.ASCII],
+  defaultEnabledEncodings: [EncodingType.BINARY, EncodingType.ASCII]
+});
+
+/**
+ * Variant 3 base level
+ * @type {NetSimLevelConfiguration}
+ */
+levels.variant3 = utils.extend(defaultLevel, {
+  showClientsInLobby: false,
+  showAddRouterButton: true,
+
+  showTabs: [
+    NetSimTabType.INSTRUCTIONS,
+    NetSimTabType.MY_DEVICE,
+    NetSimTabType.ROUTER,
+    NetSimTabType.DNS
+  ],
+
+  showEncodingControls: [EncodingType.ASCII],
+  defaultEnabledEncodings: [EncodingType.BINARY, EncodingType.ASCII],
+
+  showDnsModeControl: false,
+  defaultDnsMode: DnsMode.AUTOMATIC
+});
+
+/**
+ * AB-Sending Widget
+ * Unit 1, Lesson 4
```

Comment:
I highly recommend not referring to given levels by numbers, because once the sequence changes (new levels, levels deleted, order shuffled, etc.) the numbering gets all messed up.  I always make up a simple name for a level, and refer to it by that.